nixos/modules/misc/label.nix

```diff
@@ -11,14 +11,18 @@ in
   options.system = {
 
     nixos.label = mkOption {
-      type = types.str;
+      type = types.strMatching "[a-zA-Z0-9:_\\.-]*";
       description = ''
         NixOS version name to be used in the names of generated
         outputs and boot labels.
 
         If you ever wanted to influence the labels in your GRUB menu,
         this is the option for you.
 
+        It can't be any text, so it's expected to have only letters, numbers
+        and the following symbols: <literal>:</literal>, <literal>_</literal>,
+        <literal>.</literal> and <literal>-</literal>.
```

Comment:
```suggestion
        It can only contain letters, numbers and the following symbols:
        <literal>:</literal>, <literal>_</literal>, <literal>.</literal> and <literal>-</literal>.
```